src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -593,6 +593,23 @@ private void CompileNamedType(NamedTypeSymbol containingType)
                 }
             }
 
+            // If there is no explicit or implicit .cctor and there are static members but no initializers,
+            // then report warnings for any non-nullable fields. (If there is no .cctor, there should
+            // not be any initializers but for robustness, we check both.)
+            if (!hasStaticConstructor &&
+                members.Any(m => m.IsStatic) &&
```

Comment:
>members.Any(m => m.IsStatic) [](start = 16, length = 28)

I don't understand why all members are considered

Reply:
This is just an optimization, and not a very useful one, since `ReportUninitializedNonNullableReferenceTypeFields` shouldn't add any overhead unless there are static fields.

I'll remove this check completely. 

---
In reply to: [307945510](https://github.com/dotnet/roslyn/pull/37507#discussion_r307945510) [](ancestors = 307945510)